### PR TITLE
resource/alicloud_ess_scaling_group: Add support for protected_instances

### PR DESCRIPTION
**Pull request by TalentNing**

### CODE REVIEW

**Comment by TalentNing** [comment]
TF_ACC=1 go test ./alicloud -v -run=TestAccAlicloudEssScalingGroup -timeout=300m
=== RUN   TestAccAlicloudEssScalingGroupsDataSource
--- PASS: TestAccAlicloudEssScalingGroupsDataSource (86.43s)
=== RUN   TestAccAlicloudEssScalingGroup_basic
--- PASS: TestAccAlicloudEssScalingGroup_basic (123.79s)
=== RUN   TestAccAlicloudEssScalingGroup_desiredCapacity
--- PASS: TestAccAlicloudEssScalingGroup_desiredCapacity (67.65s)
=== RUN   TestAccAlicloudEssScalingGroup_withLaunchTemplateId
--- PASS: TestAccAlicloudEssScalingGroup_withLaunchTemplateId (57.68s)
=== RUN   TestAccAlicloudEssScalingGroup_costoptimized
--- PASS: TestAccAlicloudEssScalingGroup_costoptimized (89.48s)
=== RUN   TestAccAlicloudEssScalingGroup_vpc
--- PASS: TestAccAlicloudEssScalingGroup_vpc (98.32s)
=== RUN   TestAccAlicloudEssScalingGroup_slb
--- PASS: TestAccAlicloudEssScalingGroup_slb (124.42s)
=== RUN   TestAccAlicloudEssScalingGroup_tags
--- PASS: TestAccAlicloudEssScalingGroup_tags (56.36s)
=== RUN   TestAccAlicloudEssScalingGroup_eci
--- PASS: TestAccAlicloudEssScalingGroup_eci (54.61s)
=== RUN   TestAccAlicloudEssScalingGroup_protected_instances
--- PASS: TestAccAlicloudEssScalingGroup_protected_instances (124.00s)
PASS
ok      github.com/aliyun/terraform-provider-alicloud/alicloud  884.908s

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,7 @@
 ## 1.181.0 (Unreleased)
+
+- resource/alicloud_ess_scaling_group: Add support for protected_instances
```
Comment (xinmu0426) [inline CHANGELOG.md:3]: 181已经发布啦，辛苦修改下

File: alicloud/resource_alicloud_ess_scaling_group.go [hunk 2]
```diff
@@ -371,6 +385,12 @@ func resourceAliyunEssScalingGroupUpdate(d *schema.ResourceData, meta interface{
 		}
 		d.SetPartial("db_instance_ids")
 	}
+
+	if d.HasChange("protected_instances") {
+		oldProtectedInstances, newProtectedInstances := d.GetChange("protected_instances")
+		err = setProtectedInstances(d, client, oldProtectedInstances.(*schema.Set), newProtectedInstances.(*schema.Set))
```
Comment (xinmu0426) [inline alicloud/resource_alicloud_ess_scaling_group.go:391]: 这个快有错误的要处理下吧

File: website/docs/r/ess_scaling_group.html.markdown [hunk 3]
```diff
@@ -126,6 +126,7 @@ The following arguments are supported:
 * `tags` - (Optional, Available in v1.160.0+) A mapping of tags to assign to the resource.
   - Key: It can be up to 64 characters in length. It cannot begin with "aliyun", "acs:", "http://", or "https://". It cannot be a null string.
   - Value: It can be up to 128 characters in length. It cannot begin with "aliyun", "acs:", "http://", or "https://". It can be a null string.
+* `protected_instances` - (Optional, Available in v1.181.0+) Set or unset instances within group into protected status.
```
Comment (xinmu0426) [inline website/docs/r/ess_scaling_group.html.markdown:129]: 182啦

**Pull request merged**